test/e2e/persistent_volumes.go

```diff
@@ -66,58 +66,96 @@ var _ = framework.KubeDescribe("PersistentVolumes", func() {
+		if pvcPhase := pvc.Status.Phase; pvcPhase != "Bound" {
+			framework.Failf("Expected PersistentVolumeClaim phase Bound.  Actual phase:  %+v.  Error: %+v", pvcPhase, err)
+		}
+
+		// Checkpod runs a command to both write to the volume and read a prewritten index.html file on the volume.
+		// If both actions succeed, exits 0.  Else exits 1
+		pvc, _ = c.PersistentVolumeClaims(ns).Get(pvc.Name)
+		checkpod := makeCheckPod(ns, pvc.Name)
+		checkpod, err = c.Pods(ns).Create(checkpod)
+		if err != nil {
+			framework.Failf("Error during checkpod create: %+v", err)
 		}
 
+		// Wait for the checkpod to complete it's lifecycle
+		err = framework.WaitForPodSuccessInNamespace(c, checkpod.Name, checkpod.Spec.Containers[0].Name, checkpod.Namespace)
+		if err != nil {
+			framework.Failf("Pod failed during creation: %+v", err)
+		}
+
+		// Delete the PersistentVolumeClaim
 		framework.Logf("Deleting PersistentVolumeClaim to trigger PV Recycling")
 		err = c.PersistentVolumeClaims(ns).Delete(pvc.Name)
-		Expect(err).NotTo(HaveOccurred())
+		if err != nil {
+			framework.Failf("PersistentVolumeClaim delete experienced an unexpected error: %v", err)
+		}
 
-		// allow the recycler a chance to catch up.  it has to perform NFS scrub, which can be slow in e2e.
-		framework.WaitForPersistentVolumePhase(api.VolumeAvailable, c, pv.Name, 5*time.Second, 300*time.Second)
+		// Wait for the PersistentVolume phase to return to Available
+		framework.Logf("Waiting for recycling process to complete.")
+		err = framework.WaitForPersistentVolumePhase(api.VolumeAvailable, c, pv.Name, 3*time.Second, 300*time.Second)
+		if err != nil {
+			framework.Failf("Recycler pod experienced an unexpected error: %v", err)
+		}
 
+		// Examine the PersistentVolume.ClaimRef and UID.  Expect nil values.
 		pv, err = c.PersistentVolumes().Get(pv.Name)
-		Expect(err).NotTo(HaveOccurred())
-		if pv.Spec.ClaimRef != nil {
-			framework.Failf("Expected PersistentVolume to be unbound, but found non-nil ClaimRef: %+v", pv)
+		framework.ExpectNoError(err, "Failed to get the PersistentVolume")
+		if pv.Spec.ClaimRef != nil && len(pv.Spec.ClaimRef.UID) > 0 {
+			framework.Failf("Expected a nil ClaimRef UID", pv.Status.Phase, pv.Spec.ClaimRef.UID)
 		}
 
-		// The NFS Server pod we're using contains an index.html file
-		// Verify the file was really scrubbed from the volume
-		podTemplate := makeCheckPod(ns, serverIP)
-		checkpod, err := c.Pods(ns).Create(podTemplate)
-		framework.ExpectNoError(err, "Failed to create checker pod: %v", err)
-		err = framework.WaitForPodSuccessInNamespace(c, checkpod.Name, checkpod.Spec.Containers[0].Name, checkpod.Namespace)
-		Expect(err).NotTo(HaveOccurred())
-		// must delete PV, otherwise the PV is available and next time a PVC may bind to it and cause new PV fails to bind
+		// Delete the PersistentVolume
 		framework.Logf("Deleting PersistentVolume")
 		err = c.PersistentVolumes().Delete(pv.Name)
 		Expect(err).NotTo(HaveOccurred())
```

Comment:
Is it worth adding a `pv, err = c.PersistentVolumes().Get(pv.Name)` and expecting an error?